console-subscriber/src/builder.rs

```diff
@@ -149,6 +158,283 @@ impl Builder {
+/// for your application. If you need to add additional layers to a subscriber,
+/// see [`spawn`].
+///
+/// [default]: https://docs.rs/tracing/latest/tracing/dispatcher/index.html#setting-the-default-subscriber
+/// [sub]: https://docs.rs/tracing/latest/tracing/trait.Subscriber.html
+/// [`tracing_subscriber::fmt`]: https://docs.rs/tracing-subscriber/latest/tracing-subscriber/fmt/index.html
+/// [`Server`]: crate::Server
+///
+/// ## Configuration
+///
+/// Tokio console subscriber is configured with sensible defaults for most
+/// use cases. If you need to tune these parameters, several environmental
+/// configuration variables are available:
+///
+/// | **Environment Variable**            | **Purpose**                                                               | **Default Value** |
+/// |-------------------------------------|---------------------------------------------------------------------------|-------------------|
+/// | `TOKIO_CONSOLE_RETENTION`           | The number of seconds to accumulate completed tracing data                | 3600s (1h)        |
+/// | `TOKIO_CONSOLE_BIND`                | A HOST:PORT description, such as `localhost:1234`                         | `127.0.0.1:6669`  |
+/// | `TOKIO_CONSOLE_PUBLISH_INTERVAL`    | The number of milliseconds to wait between sending updates to the console | 1000ms (1s)       |
+/// | `TOKIO_CONSOLE_RECORD_PATH`         | The file path to save a recording                                         | None              |
+/// | `RUST_LOG`                          | Configures what events are logged events. See [`Targets`] for details.    | "error"           |
+///
+/// ## Further customization
+///
+/// To add additional layers or replace the format layer, replace
+/// `console_subscriber::init` with:
+///
+/// ```rust
+/// use tracing_subscriber::prelude::*;
+///
+/// let console_layer = console_subscriber::spawn();
+///
+/// tracing_subscriber::registry()
+///     .with(console_layer)
+///     .with(tracing_subscriber::fmt::layer())
+/// //  .with(..potential additional layer..)
+///     .init();
+/// ```
+///
+/// [`Targets`]: https://docs.rs/tracing-subscriber/latest/tracing-subscriber/filter/struct.Targets.html
+pub fn init() {
+    TasksLayer::builder().with_default_env().init();
+}
+
+/// Returns a new `tracing_subscriber` [`Layer`] configured with a [`TasksLayer`]
+/// and a [filter] that enables the spans and events required by the console.
+///
+/// This function spawns the console subscriber's [`Server`] in its own Tokio
+/// runtime in a background thread.
+///
+/// Unlike [`init`], this function does not set the default subscriber, allowing
+/// additional [`Layer`]s to be added.
+///
+/// [filter]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.Targets.html
+/// [`Layer`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Layer.html
+/// [`Server`]: crate::Server
+///
```

Comment:
nit: maybe worth adding:
```suggestion
///
/// ## Panics
///
/// * If the subscriber's background thread could not be spawned.
///
```